controllers/user/UserController.php

```diff
@@ -27,6 +29,21 @@ class UserController extends AuthBaseController
      * @return array
      */
     public function actionCreate(){
+        //Check the legal module is active or not
+        $legalModule = Yii::$app->getModule('legal');
+        if($legalModule !== null){
+            $model = new RegistrationChecks();
+            $legalKey = $model->load(Yii::$app->request->getBodyParam("legal", []), '');
+
+            //check the legal key is present or not
+            if(!$legalKey){
+                return $this->returnError(400,"Registration failed, legal module is activated but key is missing");
+            }
+            //check the dataPrivacyCheck key is true or false
+            if(!$model->dataPrivacyCheck){
+                return $this->returnError(400,"Registration failed,dataPrivacyCheck key is false");
+            }
```

Comment:
Hi @donni106 
We add a condition where we will check what inside the dataPrivacyCheck.
If the value of dataPrivacyCheck key will be = true,1,0 and false then it will be run. otherewise it will show an error (invalid value of dataPrivacyCheck key).

What is your opinion on this?

Reply:
all right, yes, good one. let's only allow `true`, `1`, `false` and `0`.